src/cryptography/hazmat/backends/openssl/ciphers.py

```diff
@@ -110,6 +112,17 @@ def update(self, data):
         if len(data) == 0 and not isinstance(self._mode, modes.GCM):
             return b""
 
+        self._bytes_processed += len(data)
+        if (
+            isinstance(self._mode, modes.GCM) and
```

Comment:
It's kind of ugly that we need to add all these special if's just for GCM. I'd be interested in seeing a solution to decoupling that. I guess we already do it in a few places here though.